src/cascadia/TerminalSettingsModel/IAppearanceConfig.idl

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "IInheritable.idl.h"
+
+#define INHERITABLE_APPEARANCE_SETTING(Type, Name) \
+    _BASE_INHERITABLE_SETTING(Type, Name);      \
+    Microsoft.Terminal.Settings.Model.IAppearanceConfig Name##OverrideSource { get; }
+
+namespace Microsoft.Terminal.Settings.Model
+{
+    [flags] enum ConvergedAlignment {
+        // low 4 bits are the horizontal
+        Horizontal_Center = 0x00,
+        Horizontal_Left = 0x01,
+        Horizontal_Right = 0x02,
+
+        // high 4 bits are the vertical
+        Vertical_Center = 0x00,
+        Vertical_Top = 0x10,
+        Vertical_Bottom = 0x20
+    };
+
+    interface IAppearanceConfig
+    {
+        INHERITABLE_APPEARANCE_SETTING(String, ColorSchemeName);

```

Comment:
The settings UI needs to be able to get the `Origin`, right? /cc @carlos-zamora

Reply:
It was the earlier implementation but that required `AppearanceConfig` to know the internals of `Profile` which we decided was not desirable